### PR TITLE
Add up command

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/103

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR

File: cli/up.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	upCmd = &cobra.Command{
+		Use:   "up <group> [scale]",
+		Short: "Bring a group up",
+		Long:  "Submit a group, with an optional scale, and start it",
+		Run:   upRun,
+	}
+)
+
+func upRun(cmd *cobra.Command, args []string) {
+	submitRun(cmd, args)
```
Comment (zeisss) [inline cli/up.go:17]: if submit raises an error, do we still continue here?

Comment (zeisss) [inline cli/up.go:17]: how does this behave for already existing slices?

Comment (JosephSalisbury) [inline cli/up.go:17]: error during submit:

```
core@core-01 ~ $ ls -R
.:
example  inagoctl

./example:
example-unit.service
core@core-01 ~ $ cat ./example/example-unit.service
[Unit]
Description=Inago Demo Unit

[Service]
ExecStart=/bin/bash -c "while true; do echo Hi; sleep 10; done"
core@core-01 ~ $ ./inagoctl up example 4
2016-03-31 10:22:23.337 | ERROR    | context.Background: [{/usr/code/.gobuild/src/github.com/giantswarm/inago/cli/submit.go:47: } {/usr/code/.gobuild/src/github.com/giantswarm/inago/cli/submit.go:80: invalid scale: must be 1 for unscalable groups}]
core@core-01 ~ $ ./inagoctl status example
2016-03-31 10:22:29.651 | ERROR    | context.Background: Failed to find group 'example'.
```

so, if submit errors and exits - which it seems to do on all errors, we're fine

for already existing slices:

```
core@core-01 ~ $ ls -R
.:
example  inagoctl

./example:
example-unit@.service
core@core-01 ~ $ cat ./example/example-unit\@.service
[Unit]
Description=Inago Demo Unit

[Service]
ExecStart=/bin/bash -c "while true; do echo Hi; sleep 10; done"
core@core-01 ~ $ ./inagoctl submit example
2016-03-31 10:20:55.241 | INFO     | context.Background: Succeeded to submit group 'example'.
core@core-01 ~ $ ./inagoctl status example
Group        Units  FDState  FCState  SAState   IP            Machine

example@b5a  *      loaded   loaded   inactive  172.17.8.103  2eb8b3bb62894c269df58a00248a68eb

core@core-01 ~ $ ./inagoctl up example
2016-03-31 10:21:09.618 | INFO     | context.Background: Succeeded to submit group 'example'.
2016-03-31 10:21:14.625 | INFO     | context.Background: Succeeded to start 2 slices for group 'example': [899 b5a].
core@core-01 ~ $ ./inagoctl status example
Group        Units  FDState   FCState   SAState  IP            Machine

example@899  *      launched  launched  active   172.17.8.102  49e7e432625f4d9fa3e5071cf3f9c26e
example@b5a  *      launched  launched  active   172.17.8.103  2eb8b3bb62894c269df58a00248a68eb
```

uncertain about the behaviour here - it is logical (it's the same as two submits, and then a start), and i'm not certain how we'd get around it anyway.

Comment (zeisss) [inline cli/up.go:17]: well, I think the expectation of `inagoctl up mygroup 4` would be that afterwards there are 4 slices. We probably can involve @marians here. 

What does this mean it there are already 2 slices? Do we just want to submit 2 slices and then start all 4?

Comment (JosephSalisbury) [inline cli/up.go:17]: If there are already 2 slices submitted of 'mygroup', and you up 2 slices of 'mygroup', then you will end up with 4 running slices, as up == submit then start, meaning that the 2 previously submitted slices will also get started.

I think having starting all four slices is logically consistent. If we see that people find this confusing, we could iterate on it - for example, up could grumble if there is an existing group.

Comment (marians) [inline cli/up.go:17]: Sounds good to me. I'm not aware of any global UX heuristic that could help us here. We have to decide and be consistent regarding this: Do we want (a) arguments like "number of slices to deal with" to affect the resulting number (add so many that we have _n_ in the end)? Or (b) should they set the number for the current command (add _n_)?

(b) is simpler, more straightforward in my opinion and also a simpler promise to keep in a distributed team.

Comment (zeisss) [inline cli/up.go:17]: > If there are already 2 slices submitted of 'mygroup', and you up 2 slices of 'mygroup', then you will end up with 4 running slices, as up == submit then start, meaning that the 2 previously submitted slices will also get started.

this sounds like the current implementation, but what would we expect "up" to do? If I do `inago up mygroup 4` and two are already there, do I end up with 6 or 4?

Comment (JosephSalisbury) [inline cli/up.go:17]: I think (b) is more in line with the current design, as it is now.

Comment (JosephSalisbury) [inline cli/up.go:17]: @zeisss sorry, i misunderstood, that is the current implementation.

I think if I `up`'d 4, I would expect four to be there, but I don't think it's surprising that if you submit 2 of the same group beforehand, they also get started.

Comment (zeisss) [inline cli/up.go:17]: ack

Comment (zeisss) [inline cli/up.go:17]: ack, I also would expect: "up 4" makes the end number 4, all started.

Comment (JosephSalisbury) [inline cli/up.go:17]: so, if there were already 2 submitted units, you'd expect 2 units to be started, and then all 4 started?

i don't feel like this is consistent with the current behaviour of submit - submit 3 will _add_ 3, while it sounds like up 3 will _make sure there are 3_

Comment (zeisss) [inline cli/up.go:17]: ack, but thats just my expectation. thats why should talk with @marians

Comment (JosephSalisbury) [inline cli/up.go:17]: if i understand https://github.com/giantswarm/inago/pull/195#discussion_r58044528 correctly, the current implementation sounds good to @marians, assuming we are consistent with it (in any later commands)

Comment (zeisss) [inline cli/up.go:17]: k :)

**Comment by zeisss** [comment]
LGTM

**Pull request merged**